external-providers/dotnet-external-provider/pkg/dotnet/service_client.go

```diff
@@ -59,13 +60,68 @@ func (d *dotnetServiceClient) Evaluate(ctx context.Context, cap string, conditio
 		return provider.ProviderEvaluateResponse{}, fmt.Errorf("unable to get namespace for query")
 	}
 
+	// Use FileSearcher to properly handle all filepath filtering sources:
+	// 1. Provider config constraints (from InitConfig)
+	// 2. Rule scope constraints (from GetScopedFilepaths)
+	// 3. Condition filepaths (if they exist in the condition)
+	includedFilepaths, excludedFilepaths := cond.ProviderContext.GetScopedFilepaths()
+
+	// Only use FileSearcher if we have any filepath constraints
+	fileMap := make(map[string]struct{})
+	if len(includedFilepaths) > 0 || len(excludedFilepaths) > 0 {
+		fileSearcher := provider.FileSearcher{
+			BasePath: d.config.Location,
+			ProviderConfigConstraints: provider.IncludeExcludeConstraints{
+				IncludePathsOrPatterns: provider.GetIncludedPathsFromConfig(d.config, true),
+				ExcludePathsOrPatterns: provider.GetExcludedDirsFromConfig(d.config),
+			},
+			RuleScopeConstraints: provider.IncludeExcludeConstraints{
+				IncludePathsOrPatterns: includedFilepaths,
+				ExcludePathsOrPatterns: excludedFilepaths,
+			},
+			FailFast: true,
+			Log:      d.log,
+		}
+
+		// Run file search in a goroutine to build allowed file map
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// Note: dotnet conditions don't have a Filepaths field
+			paths, err := fileSearcher.Search(provider.SearchCriteria{
+				ConditionFilepaths: []string{},
+			})
+			if err != nil {
+				d.log.Error(err, "failed to search for files")
+				return
+			}
+			for _, path := range paths {
+				normalizedPath := provider.NormalizePathForComparison(path)
+				fileMap[normalizedPath] = struct{}{}
+			}
+		}()
+
+		// Wait for file search to complete
+		wg.Wait()
+	}
+
 	symbols := d.GetAllSymbols(query)
 	incidents := []provider.IncidentContext{}
 	for _, s := range symbols {
 		if s.Kind == protocol.SymbolKindMethod {
 			references := d.GetAllReferences(s)
 			for _, ref := range references {
 				if strings.Contains(ref.URI.Filename(), d.config.Location) {
+					normalizedRefPath := provider.NormalizePathForComparison(string(ref.URI))
+
+					// If FileSearcher found specific files, filter to only those files
+					if len(fileMap) > 0 {
```

Comment:
@pranavgaikwad would have to check me on this, but I think it may be valid that we don't find any matching files